### PR TITLE
Search/charm error messages

### PR DESCRIPTION
**Pull request by huwshimi**
Display error messages if there is an error loading the search results or an entity details.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- modify the following lines to match be `'nothing-here'`:
- https://github.com/huwshimi/juju-gui/blob/search-results-errors/jujugui/static/gui/src/app/components/entity-details/entity-details.js#L132
https://github.com/huwshimi/juju-gui/blob/search-results-errors/jujugui/static/gui/src/app/components/search-results/search-results.js#L187
- open the mid-point and click on a charm
- you should get an error message
- do a search
- you should get an error message

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3565/
Test PASSed.

**Comment by hatched** [comment]
When you get the error message it kind of traps the user there. They have to either search or close sectionC. Is a 'back' step possible here?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3574/
Test PASSed.

**Comment by huwshimi** [comment]
@hatched back links added.

File: jujugui/static/gui/src/app/components/entity-details/test-entity-details.js [hunk 1]
```diff
@@ -79,18 +79,57 @@ describe('EntityDetails', function() {
...
+          pluralize={pluralize} />, true);
+    shallowRenderer.getMountedInstance().componentDidMount();
+    var output = shallowRenderer.getRenderOutput();
+    var expected = (
+      <div className="entity-details">
+        <p className="error">
+          There was a problem while loading the entity details.
+          You could try searching for another charm or bundle or go{' '}
+          <span className="link"
+            onClick={output.props.children.props.children[2].props.onClick}>
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/entity-details/test-entity-details.js]: What about doing the `instance.actualMethod` trick for these tests?

**Comment by hatched** [comment]
Looks awesome! Thanks! :+1: QA :ok:

File: jujugui/static/gui/src/app/components/entity-details/test-entity-details.js (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/entity-details/test-entity-details.js]: Great idea, done.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3576/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3577/
Test PASSed.

**Pull request closed without merging**